### PR TITLE
Refine check for existence of `prom_schema_migrations` table

### PR DESCRIPTION
**Pull request by JamesGuthrie**
## Description

The table could exist in any schema. We expect there to be either 0 or 1
tables. If there are more then something really weird has happened, and
we can't safely proceed.

Part of: https://github.com/timescale/promscale_extension/issues/371

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [ ] ~~Updated the relevant documentation~~

### CODE REVIEW

**Review by jgpruitt — APPROVED** [review]

File: pkg/tests/end_to_end_tests/new_migrate_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"context"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stretchr/testify/require"
+	"github.com/timescale/promscale/pkg/pgmodel"
+	"github.com/timescale/promscale/pkg/pgmodel/common/extension"
+	"github.com/timescale/promscale/pkg/version"
+	"testing"
+)
+
+func TestMigrationFailesIfMultiplePromSchemaMigrationsTables(t *testing.T) {
```
Comment (antekresic) [inline pkg/tests/end_to_end_tests/new_migrate_test.go:16]: Nit: test would be easier to read/understand if you were to run migrate once, add an additional schema table and check that migrate fails with the correct error.

**Comment by niksajakovljevic** [comment]
Why don't we define `prom_schema_migrations` in a specific schema? That could help us avoid this code completly

File: pkg/pgmodel/new_migrate.go [hunk 2]
```diff
@@ -23,12 +23,31 @@ var (
...
+	}
+
+	switch len(schemas) {
+	case 0:
+		return false, nil
+	case 1:
+		return true, nil
+	default:
+		return false, fmt.Errorf("`prom_schema_migrations` exists in multiple schemas: %s", schemas)
+	}
```
Comment (antekresic) [inline pkg/pgmodel/new_migrate.go:50]:
```suggestion
        if len(schemas) > 1 {
            return false, fmt.Errorf("`prom_schema_migrations` exists in multiple schemas: %s", schemas)
        }
        return len(schemas) == 1, nil
```

**Review by antekresic — CHANGES_REQUESTED** [review]

File: pkg/tests/end_to_end_tests/new_migrate_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		extOptions := extension.ExtensionMigrateOptions{Install: true, Upgrade: true, UpgradePreRelease: true}
+
+		err = pgmodel.Migrate(conn.Conn(), pgmodel.VersionInfo{Version: version.Promscale, CommitHash: "azxtestcommit"}, nil, extOptions)
+
+		require.Equal(t, err.Error(), "failed to determine whether the prom_schema_migrations table existed: `prom_schema_migrations` exists in multiple schemas: [public myschema]")
```
[the remark this answers is not in the document]
Comment (JamesGuthrie) [reply pkg/tests/end_to_end_tests/new_migrate_test.go:43]: As discussed, I will ignore this suggestion.

File: pkg/pgmodel/new_migrate.go (hunk 2 above)
Comment (JamesGuthrie) [reply pkg/pgmodel/new_migrate.go:50]: To be honest I prefer the way it is. To me it's simpler to read. I don't really mind either way though.

File: pkg/tests/end_to_end_tests/new_migrate_test.go (hunk 1 above)
Comment (JamesGuthrie) [reply pkg/tests/end_to_end_tests/new_migrate_test.go:16]: I've been fiddling with this for a while, and to be honest there's no easy way to do what you're asking for. As you brought it up as a nit, I'm not sure if it's worth sinking more time into this. I did find a couple of small improvements to this test, which I will push.

Comment (JamesGuthrie) [reply pkg/tests/end_to_end_tests/new_migrate_test.go]: I did end up figuring out how to do this and implemented it as suggested.

**Review by antekresic — APPROVED** [review]

**Pull request merged**